### PR TITLE
Hotfix/miner builder 1.11

### PR DESCRIPTION
**Pull request by Raycoms**
Merge all the newest 1.10 stuff to 1.11

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:229]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:264]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:280]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: MINOR Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:239]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @javax.annotation.Nullable
+    @Deprecated
+    @Override
+    public AxisAlignedBB getCollisionBoundingBox(final IBlockState blockState, final IBlockAccess worldIn, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:192]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+                                      (float) E_END_COLLISION_X,
+                                      (float) HEIGHT_COLLISION,
+                                      (float) E_END_COLLISION_Z);
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @NotNull
+    @Override
+    public IBlockState getStateForPlacement(
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:323]: MINOR Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java:211]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java [hunk 9]
```diff
@@ -358,7 +358,7 @@ public BlockPos getPositionOfChestWithTool(@NotNull final String tool, final int
             for (@NotNull final BlockPos pos : building.getAdditionalCountainers())
             {
                 final TileEntity entity = world.getTileEntity(pos);
-                if (entity instanceof TileEntityChest && ((minLevel != -1 && InventoryUtils.isPickaxeInTileEntity((TileEntityChest) entity, minLevel, building.getBuildingLevel()))
+                if (entity instanceof TileEntityChest && ((minLevel != -1 && InventoryUtils.isPickaxeInTileEntity((TileEntityChest) entity, minLevel, requestingBuilding.getBuildingLevel()))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java:361]: MINOR Split this 189 characters long line (which is greater than 180 authorized). [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/commands/AbstractCitizensCommands.java [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <CitizenId>";
+    }
+
+    @NotNull
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AbstractCitizensCommands.java:44]: MAJOR The Cyclomatic Complexity of this method "execute" is 11 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/commands/DeleteColonyCommand.java [hunk 11]
```diff
@@ -45,46 +50,72 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/DeleteColonyCommand.java:51]: MAJOR Reduce the number of returns of this method 6, down to the maximum allowed 5. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/commands/AddOfficerCommand.java [hunk 12]
```diff
@@ -44,10 +50,49 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AddOfficerCommand.java:51]: MAJOR The Cyclomatic Complexity of this method "execute" is 12 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 13]
```diff
@@ -28,6 +28,10 @@ private RecipeHandler()
      */
     public static void init(final boolean enableInDevelopmentFeatures, final boolean supplyChests)
     {
+        GameRegistry.addRecipe(new ItemStack(ModBlocks.blockConstructionTape, 16), "SWS", "S S", "S S",
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java:31]: MINOR Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/commands/RandomTeleportCommand.java [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+     * Method used to teleport the player.
+     *
+     * @param sender           the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private static void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/RandomTeleportCommand.java:126]: MAJOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS135)

File: src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java [hunk 15]
```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
...
+    public boolean isPlayerOpped(@NotNull final ICommandSender sender, String cmdName)
+    {
+        int requiredOpLevel = PERMNUM;
+        if (PERMNUM < 1)
+        {
+            requiredOpLevel = 1;
+        }
+        if (PERMNUM > 4)
+        {
+            requiredOpLevel = 4;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:160]: MINOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java [hunk 16]
```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
...
+        return colony.getPermissions().getRank(player).equals(Permissions.Rank.OFFICER) || colony.getPermissions().getRank(player).equals(Permissions.Rank.OWNER);
+    }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     *
+     * @param theCommand which command to check if the player can use it
+     * @return boolean
+     */
+    public boolean canCommandSenderUseCommand(Commands theCommand)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:116]: MAJOR Reduce the number of returns of this method 10, down to the maximum allowed 5. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java [hunk 17]
```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
...
+    {
+        switch (theCommand)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:118]: MAJOR Complete cases by adding the missing enum constants or add a default case to this switch. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

File: src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java [hunk 18]
```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
...
+     */
+    @NotNull
+    public boolean isPlayerOpped(@NotNull final ICommandSender sender, String cmdName)
+    {
+        int requiredOpLevel = PERMNUM;
+        if (PERMNUM < 1)
+        {
+            requiredOpLevel = 1;
+        }
+        if (PERMNUM > 4)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java:158]: MINOR Assign this magic number 4 to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 19]
```diff
@@ -94,6 +94,11 @@
     private WalkToProxy proxy;
 
     /**
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:96]: MINOR First sentence should end with a period. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 20]
```diff
@@ -171,8 +176,35 @@ protected AbstractBuildingWorker getOwnBuilding()
...
+                final String jobName = colonyJob == null ? "null" : colonyJob.getName();
+                Log.getLogger().error("Pausing Entity " + name + " (" + jobName + ") at " + workerPosition + " for " + timeout + " Seconds because of error:");
+            }
+            else
+            {
+                Log.getLogger().error("Pausing Entity that is null for " + timeout + " Seconds because of error:");
+            }
+
+            // fix for printing the actual exception
+            e.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:199]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 21]
```diff
@@ -171,8 +176,35 @@ protected AbstractBuildingWorker getOwnBuilding()
...
     {
-        Log.getLogger().info("Pausing Entity for 5 Seconds");
-        this.setDelay(EXCEPTION_TIMEOUT);
+        try
+        {
+            final int timeout = EXCEPTION_TIMEOUT * exceptionTimer;
+            this.setDelay(timeout);
+            // wait for longer now
+            exceptionTimer *= 2;
+            if (worker != null)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:185]: MAJOR Avoid if (x != y) ..; else ..; [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 22]
```diff
@@ -171,8 +176,35 @@ protected AbstractBuildingWorker getOwnBuilding()
...
+                Log.getLogger().error("Pausing Entity that is null for " + timeout + " Seconds because of error:");
+            }
+
+            // fix for printing the actual exception
+            e.printStackTrace();
+        }
+        catch (RuntimeException exp)
+        {
+            Log.getLogger().error("Welp reporting crashed:");
+            exp.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:204]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 23]
```diff
@@ -171,8 +176,35 @@ protected AbstractBuildingWorker getOwnBuilding()
...
+            Log.getLogger().error("Caused by ai exception:");
+            e.printStackTrace();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java:206]: CRITICAL Use a logger to log this exception. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/com/minecolonies/coremod/commands/ChangeColonyOwnerCommand.java [hunk 24]
```diff
@@ -0,0 +1,145 @@
...
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <(Optional)Player>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/ChangeColonyOwnerCommand.java:51]: MAJOR Reduce the number of returns of this method 9, down to the maximum allowed 5. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
MAJOR The Cyclomatic Complexity of this method "execute" is 14 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java [hunk 25]
```diff
@@ -0,0 +1,292 @@
...
+            placeConstructionTape(new WorkOrderBuild(building, 1), world);
+    }
+
+    /**
+     * Place construction tape.
+     * @param workOrder the workorder.
+     * @param world the world.
+     */
+
+    public static void placeConstructionTape(@NotNull WorkOrderBuild workOrder, @NotNull World world)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java:109]: MAJOR The Cyclomatic Complexity of this method "placeConstructionTape" is 11 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 26]
```diff
@@ -0,0 +1,273 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:198]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 27]
```diff
@@ -0,0 +1,273 @@
...
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @javax.annotation.Nullable
+    @Deprecated
+    @Override
+    public AxisAlignedBB getCollisionBoundingBox(final IBlockState blockState, final IBlockAccess worldIn, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:151]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 28]
```diff
@@ -0,0 +1,273 @@
...
+                                      (float) SN_END_COLLISION_X,
+                                      (float) HEIGHT_COLLISION,
+                                      (float) SN_END_COLLISION_Z);
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @NotNull
+    @Override
+    public IBlockState getStateForPlacement(
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:265]: MINOR Don't override a deprecated method or explicitly mark it as "@Deprecated". [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 29]
```diff
@@ -0,0 +1,273 @@
...
+
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:170]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 30]
```diff
@@ -0,0 +1,273 @@
...
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: MINOR Remove this unused import 'org.jetbrains.annotations.Nullable'. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 31]
```diff
@@ -0,0 +1,273 @@
...
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:188]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 32]
```diff
@@ -0,0 +1,273 @@
...
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:223]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 33]
```diff
@@ -0,0 +1,273 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java:239]: INFO Do not forget to remove this deprecated code someday. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1133)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 36 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 10 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [AbstractBuilding.java](https://github.com/Minecolonies/minecolonies/blob/3bdb4b891578ed2f2011eef958ecf6f1a141b5d9/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 35 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
We have to ignore the sonarQube failure since we don't want to log these errors directly.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 33 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 11 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: and this too, gogogog

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/582/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 32 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 12 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Pull request merged**